File: src/lib/api/noteAi.ts
/**
 * 笔记轻量 AI 产物的写入 API（B1 ＋轻量 AI）。
 *
 * 对应 src-tauri/src/commands/note_ai.rs。
 *
 * 🔴 **这里不调模型**：模型调用走现成的变换枢纽（`getTransform("ai-summarize").run`），
 * 那条路上已经有 `ai_enabled` 门控、出网闸、预算、缓存与用量日志。
 * 本文件只管把拿到的结果写回库里。
 */
import { invoke } from "@tauri-apps/api/core";
import { logger } from "@/lib/logger";
import { toastActionFailed } from "@/lib/utils";

/** 写一行摘要。传 null = 清掉。 */
export async function noteSetSummary(id: string, summary: string | null): Promise<boolean> {
  try {
    await invoke("note_set_summary", { id, summary });
    return true;
  } catch (e) {
    logger.error("保存摘要失败", e);
    toastActionFailed("保存摘要", e);
    return false;
  }
}

/**
 * 把模型输出里的标签追加给笔记，返回**真正新增**的那几个。
 * 解析在后端（容错规则多，需要单测），前端只把原文传过去。
 */
export async function noteAddAiTags(noteId: string, raw: string): Promise<string[] | null> {
  try {
    return await invoke<string[]>("note_add_ai_tags", { noteId, raw });
  } catch (e) {
    logger.error("写入 AI 标签失败", e);
    toastActionFailed("写入标签", e);
    return null;
  }
}

/** 用户确认：把这篇笔记的 AI 标签转成手动标签。 */
export async function noteConfirmAiTags(noteId: string): Promise<boolean> {
  try {
    await invoke("note_confirm_ai_tags", { noteId });
    return true;
  } catch (e) {
    logger.error("确认 AI 标签失败", e);
    toastActionFailed("确认标签", e);
    return false;
  }
}

File: src-tauri/src/commands/note_ai.rs
//! 笔记轻量 AI 产物的写入命令（B1 ＋轻量 AI）。
//!
//! 🔴 **这里不调模型，也不判 `ai_enabled`**。前端先走 `ai_run`（那条路上有
//! 开关门控、出网闸、预算、缓存、用量日志），拿到输出后再调这里落库。
//!
//! 为什么不把两步合成一个命令：`ai_run` 会返回「需要确认」与「超预算」这两种
//! **不是错误的正常结果**，得由前端分支处理（弹确认框 / 提示预算）。
//! 包成一个命令就得把那套状态机在后端再造一遍。

use crate::data_store::DataStore;
use tauri::State;

/// 写一行摘要。`summary` 为 `None` = 清掉已有摘要。
#[tauri::command]
pub fn note_set_summary(
    store: State<DataStore>,
    id: String,
    summary: Option<String>,
) -> Result<(), String> {
    store.note_set_summary(&id, summary.as_deref())
}

/// 把模型输出里的标签追加给笔记（标为 `source='ai'`），返回真正新增的那几个。
///
/// 解析在后端（`parse_ai_tags`）而不是前端：它是纯函数、容错规则多，放后端才能写单测。
#[tauri::command]
pub fn note_add_ai_tags(
    store: State<DataStore>,
    note_id: String,
    raw: String,
) -> Result<Vec<String>, String> {
    store.note_add_ai_tags(&note_id, &raw)
}

/// 用户确认：把这篇笔记的 AI 标签转成手动标签。
#[tauri::command]
pub fn note_confirm_ai_tags(store: State<DataStore>, note_id: String) -> Result<(), String> {
    store.note_confirm_ai_tags(&note_id)
}
